File: js/validAnagram.js
function isAnagram(s, t) {
  if (s.length !== t.length) {
    return false;
  }
  const x = {};
  for (let letter of s) {
    x[letter] = x[letter] ? (x[letter] += 1) : 1;
  }
  for (let letter of t) {
    if (x[letter] === 0 || !x[letter]) {
      return false;
    }
    x[letter]--;
  }
  return true;
}

File: python/valid_anagram.py
def is_anagram(s, t):
    if len(s) != len(t):
        return False
    counter = {}
    for letter in s:
        counter[letter] = counter.get(letter, 0) + 1
    for letter in t:
        if letter not in counter or counter[letter] == 0:
            return False
        counter[letter] -= 1
    return True
